CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Camera/CameraPreviewViewController.swift

```diff
@@ -366,6 +370,48 @@ final class CameraPreviewViewController: UIViewController {
     func changeCameraFrameColor(to color: UIColor) {
         cameraFrameView.image = cameraFrameView.image?.tintedImageWithColor(color)
     }
+
+    // MARK: - IBAN detection
+    private var textOrientation = CGImagePropertyOrientation.up
+    private var currentOrientation: UIInterfaceOrientation = .portrait
+    // Vision -> AVF coordinate transform.
+    private var visionToAVFTransform = CGAffineTransform.identity
+
+    private func setupOrientationAndTransform() {
+        // MARK: - Coordinate transforms
+        // Transform from UI orientation to buffer orientation.
+        var uiRotationTransform = CGAffineTransform.identity
+        // Transform bottom-left coordinates to top-left.
+        let bottomToTopTransform = CGAffineTransform(scaleX: 1, y: -1).translatedBy(x: 0, y: -1)
+        // Transform coordinates in ROI to global coordinates (still normalized).
+        var roiToGlobalTransform = CGAffineTransform.identity
+        // Region of video data output buffer that recognition should be run on.
+        // Gets recalculated once the bounds of the preview layer are known.
+        let roi = CGRect(x: 0, y: 0, width: 1, height: 1)
+        // Recalculate the affine transform between Vision coordinates and AVF coordinates.
+        roiToGlobalTransform = CGAffineTransform(translationX: roi.origin.x, y: roi.origin.y)
+            .scaledBy(x: roi.width, y: roi.height)
+
+        // Compensate for orientation (buffers always come in the same orientation).
+        switch currentOrientation {
```

Comment:
[Suggestion]: Could you please extract switch case into a separate method?

Reply:
I don't see the purpose in extracting it outside of this method since it sets some properties that are important just in this method and not in any other part. 
Also, that is the purpose of this method to set the correct text orientation and do the transformation.

Second thing is that I will prefer to keep this code in one method since for me is not very clear what are all the transformations done here and not spread it everywhere.

Also, I'm planning to check all the orientation properties that we set in this class when I will work to fix the bug for iPad(https://ginis.atlassian.net/browse/PIA-4627).

Reply:
Agree!